magnum-kyo/src/main/scala/com/augustnagro/magnum/magkyo/TransactorKyo.scala

```diff
@@ -0,0 +1,201 @@
+package com.augustnagro.magnum.magkyo
+
+import com.augustnagro.magnum.{DbCon, DbTx, SqlException, SqlLogger}
+import kyo.*
+
+import java.sql.Connection
+import javax.sql.DataSource
+import scala.util.control.NonFatal
+
+class TransactorKyo private (
+    dataSource: DataSource,
+    sqlLogger: SqlLogger,
+    connectionConfig: Connection => Unit,
+    semaphore: Maybe[Meter]
+):
+  def withSqlLogger(sqlLogger: SqlLogger): TransactorKyo =
+    new TransactorKyo(
+      dataSource,
+      sqlLogger,
+      connectionConfig,
+      semaphore
+    )
+
+  def withConnectionConfig(
+      connectionConfig: Connection => Unit
+  ): TransactorKyo =
+    new TransactorKyo(
+      dataSource,
+      sqlLogger,
+      connectionConfig,
+      semaphore
+    )
+
+  def connect[A](f: DbCon ?=> A): A < (Abort[Throwable] & Async) =
+    val res =
+      Resource
+        .acquireRelease(acquireConnection)(releaseConnection)
+        .map { lo =>
+          IO {
+            connectionConfig(lo)
+            f(using DbCon(lo, sqlLogger))
+          }
+        }
+    val effect = Resource.run(res)
+    semaphore.fold(effect)(_.run(effect))
+
+  def transact[A](f: DbTx ?=> A): A < (Abort[Throwable] & Async) =
+    val res =
+      Resource
+        .acquireRelease(acquireConnection)(releaseConnection)
+        .map { lo =>
+          Async.mask {
+            connectionConfig(lo)
+            lo.setAutoCommit(false)
+            Abort
+              .catching[Throwable](
+                f(using DbTx(lo, sqlLogger))
+              )
+              .foldAbort(
+                out =>
+                  IO {
+                    lo.commit()
+                    out
+                  },
+                error =>
+                  IO {
+                    lo.rollback()
+                    Abort.fail(error)
+                  }
+              )
+          }
+        }
+    val effect = Resource.run(res)
+    semaphore.fold(effect)(_.run(effect))
+  end transact
+
+  private val acquireConnection: Connection < (IO & Abort[SqlException]) =
+    IO(
+      Abort
+        .catching[Throwable](dataSource.getConnection())
+        .mapAbort[SqlException, Any](t =>
+          SqlException("Unable to acquire DB Connection", t)
+        )
+    )
+
+  private def releaseConnection(con: Connection) =
+    if con eq null then ()
+    else
+      IO(
+        Abort
+          .catching[Throwable](con.close())
+          .mapAbort[SqlException, Any](t =>
+            SqlException("Unable to close DB Connection", t)
+          )
+          .orPanic
+      )
```

Comment:
I don't think this error will be seen by users. Can you try:
```suggestion
      IO(con.close())
```

Reply:
is this specifically for release, do we need to worry about the other places?